### PR TITLE
feat(HIP): HIP-646 and HIP-765

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:
This PR implements HIP-646 and HIP-765

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-sdk-go/issues/777 https://github.com/hashgraph/hedera-sdk-go/issues/885

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: examples/token_update_metadata/main.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	fmt.Println("create admin key: ", adminKey.String())
+
+	var initialMetadata = []byte{1, 2, 3}
+	var newMetadata = []byte{3, 4, 5, 6}
+
+	tx, err := hedera.NewTokenCreateTransaction().
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetDecimals(3).
+		SetTokenType(hedera.TokenTypeFungibleCommon). // The same flow can be executed with a TokenTypeNonFungibleUnique (i.e. HIP-765)
```
Comment (gsstoykov) [inline examples/token_update_metadata/main.go:52]: The example could accept the token type like `-ft`, `-nft` and here you can just put in the token type after the token type is resolved.

**Review by gsstoykov — APPROVED** [review]
LGTM.

File: token_create_transaction.go [hunk 2]
```diff
@@ -377,6 +393,18 @@ func (tx *TokenCreateTransaction) GetMetadataKey() Key {
 	return tx.metadataKey
 }
 
+// SetMetadata Sets the metadata for the token
+func (tx *TokenCreateTransaction) SetMetadata(metadata []byte) *TokenCreateTransaction {
```
Comment (0xivanov) [inline token_create_transaction.go:397]: Remove these 2 methods as we already have GetTokenMetadata and SetTokenMetadata

File: examples/token_update_nfts/main.go [hunk 3]
```diff
@@ -135,7 +134,7 @@ func main() {
 		panic(fmt.Sprintf("%v : error getting receipt", err))
 	}
 
-	// Update nfts metadata
+	// update nfts metadata
```
Comment (0xivanov) [inline examples/token_update_nfts/main.go:137]: Revert

**Review by gsstoykov — APPROVED** [review]

**Pull request merged**